src/Console/Arguments.php

```diff
@@ -0,0 +1,84 @@
+<?php
+/**
+ * CakePHP(tm) : Rapid Development Framework (https://cakephp.org)
+ * Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ *
+ * Licensed under The MIT License
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (https://cakefoundation.org)
+ * @link          https://cakephp.org CakePHP(tm) Project
+ * @since         3.6.0
+ * @license       https://opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Cake\Console;
+
+/**
+ * Provides an interface for interacting with
+ * a command's options and arguments.
+ */
+class Arguments
+{
+    /**
+     * Positional arguments.
+     *
+     * @var array
+     */
+    protected $args;
+
+    /**
+     * Named options
+     *
+     * @var array
+     */
+    protected $options;
+
+    /**
+     * Constructor
+     *
+     * @param array $args Positional arguments
+     * @param array $options Named arguments
+     */
+    public function __construct(array $args, array $options)
```

Comment:
Should it maybe have the ability to name arguments? like being able to address the 3rd argument by name.
I've found this to be very useful in other frameworks.

Reply:
I would then use a similar approach of naming as I mentioned in the main ticket:

    getArgumentAt(index)
    hasArgumentAt(index)

and the new

    hasArgument(string)

maybe then.

Reply:
@lorenzo Are you thinking of a map between names and indexes?

Reply:
Yes, exactly

Reply:
Ok. I was thinking of skipping that. But if you think it will be valuable to have. I'll add it in.

Reply:
I've renamed the methods to include 'At'. I'll add the name based methods in a subsequent pull request.